src/components/controller/controller.ts

```diff
@@ -0,0 +1,43 @@
+import AppLoader from './appLoader';
+import { NewsSourceResponse, NewsAPIResponse } from '../types';
+
+class AppController extends AppLoader {
+    getSources(callback: (data: NewsSourceResponse) => void): void {
+        super.getResp(
+            {
+                endpoint: 'sources',
+            },
+            callback
+        );
+    }
+
+    getNews(e: MouseEvent, callback: (data: NewsAPIResponse) => void): void {
+        let target = e.target as HTMLElement | null;
```

Comment:
Please, try to avoid as. You can use instanceof to check type, or generic
```
if (!(e.currentTarget instanceof HTMLElement)) {
            throw new Error('newsContainer is not an HTMLElement');
        }

        if (!(e.target instanceof HTMLElement)) {
            return;
        }

        const newsContainer = e.currentTarget;
        let current: HTMLElement | null = e.target;
        
        ...
}
```